src/ucp/core/ucp_ep.c

```diff
@@ -1615,45 +1618,52 @@ ucs_status_t ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config,
+                ucp_ep_config_set_rndv_thresh(worker, config, rkey_ptr_lanes,
+                                              iface_attr->cap.get.min_zcopy,
+                                              SIZE_MAX, &config->rndv.rma_thresh);
+            } else {
+                ucp_ep_config_set_rndv_thresh(worker, config,
+                                              config->key.rma_bw_lanes,
+                                              iface_attr->cap.get.min_zcopy,
+                                              SIZE_MAX, &config->rndv.rma_thresh);
+            }
+
+            ucp_ep_config_set_am_rndv_thresh(worker, iface_attr, md_attr, config,
+                                             iface_attr->cap.am.min_zcopy,
+                                             SIZE_MAX, &config->rndv.am_thresh);
+
             /* All keys must fit in RNDV packet.
              * TODO remove some MDs if they don't
              */
-            ucs_assert_always(config->tag.rndv.rkey_size <= config->am.max_bcopy);
+            ucs_assert_always(config->rndv.rkey_size <= config->am.max_bcopy);
 
             if (!ucp_ep_is_tag_offload_enabled(config)) {
                 /* Tag offload is disabled, AM will be used for all
                  * tag-matching protocols */
                 /* TODO: set threshold level based on all available lanes */
 
-                config->tag.eager  = config->am;
-                config->tag.lane   = lane;
-                min_rndv_thresh    = iface_attr->cap.get.min_zcopy;
-                min_am_rndv_thresh = iface_attr->cap.am.min_zcopy;
-
-                if (config->key.rkey_ptr_lane != UCP_NULL_LANE) {
-                    rkey_ptr_lanes[0] = config->key.rkey_ptr_lane;
-                    ucp_ep_config_set_rndv_thresh(worker, config,
-                                                  rkey_ptr_lanes,
-                                                  min_rndv_thresh,
-                                                  max_rndv_thresh);
-                } else {
-                    ucp_ep_config_set_rndv_thresh(worker, config,
-                                                  config->key.rma_bw_lanes,
-                                                  min_rndv_thresh,
-                                                  max_rndv_thresh);
+                config->tag.eager           = config->am;
+                config->tag.lane            = lane;
+                config->tag.rndv.am_thresh  = config->rndv.am_thresh;
+                config->tag.rndv.rma_thresh = config->rndv.rma_thresh;
+
+                if (context->config.ext.rndv_thresh != UCS_MEMUNITS_AUTO) {
+                    /* adjust max_short if rndv_thresh is set externally */
+                    min_rndv_thresh = ucs_min(config->tag.rndv.rma_thresh.remote,
+                                              config->tag.rndv.am_thresh.remote);
+                    ucp_ep_config_adjust_max_short(&config->tag.eager.max_short,
```

Comment:
is this fixing a bug with eager and max_short? i didn't see it the code before change

Reply:
no, we adjust max short, overwritten on line 1649.

Reply:
@yosefe, do you have other comments?

Reply:
per offline discussion - it was added because set_rndv_thresh was moved to earlier phase